fix(docs): Updating info in aztec-nr docs

critesjosh opened the pull request:
## Summary

Comprehensive documentation review and improvements for the aztec-nr developer documentation. This PR focuses on improving clarity, structure, accuracy, and consistency across the Aztec smart contract development guides.

### Changes Include:

**Framework & Core Concepts**
- Improved storage declaration guide with clearer examples and structure
- Enhanced macros documentation for better clarity
- Updated contract artifacts, structure, upgrades, and dependencies documentation
- Refined global variables documentation

**Function Documentation**
- Updated attributes documentation for clarity and accuracy
- Streamlined function transforms and visibility documentation

**How-To Guides**
- Enhanced event emission guide with clearer structure and examples
- Refined custom notes implementation guide
- Streamlined how-to guide for calling contracts
- Updated authwit (authentication witness) guide for clarity
- Improved cross-chain communication guide

**Advanced Topics**
- Enhanced documentation for proving historical state
- Improved retrieving and filtering notes guide
- Updated partial notes and capsules documentation
- Refactored profiling transactions guide

**L1-L2 Messaging**
- Updated L1-L2 messaging data structures documentation
- Improved inbox, outbox, and registry documentation

**Setup & Installation**
- Updated installation instructions for Noir LSP to specify VSCode Extension
- Improved local network installation instructions

**General Improvements**
- Fixed grammatical errors and typos throughout
- Improved consistency in tone and terminology
- Updated links and cross-references
- Removed redundant information and improved conciseness
- Better organization of content following simple → complex flow

sklppy88 commented on docs/docs-developers/docs/aztec-nr/framework-description/advanced/how_to_retrieve_filter_notes.md:
```diff
@@ -26,9 +32,21 @@ This returns up to `MAX_NOTE_HASH_READ_REQUESTS_PER_CALL` notes without filterin
 
+:::tip get_notes vs pop_notes
+
+- `get_notes`: Retrieves notes without nullifying. Use when you need to read note data.
```
Note data is not guaranteed to be current, or non-nullified

sklppy88 commented on docs/docs-developers/docs/aztec-nr/framework-description/how_to_implement_custom_notes.md:
As an aside, I think we should move everything to include codes. Right now it's a mix of code blocks and include codes.

critesjosh replied:
i agree. I updated a bunch, let's tackle the rest in a future pr

sklppy88 commented on docs/docs-developers/docs/aztec-nr/framework-description/ethereum-aztec-messaging/index.md:
```diff
@@ -109,32 +109,14 @@ For the sake of cross-chain messages, this means inserting and nullifying L1 $\r
+The message structures are:
+- `L1Actor` and `L2Actor` - Represent entities on L1 and L2 respectively
+- `L1ToL2Msg` - Message sent from L1 to L2
+- `L2ToL1Msg` - Message sent from L2 to L1
```
This section a bit verbose

sklppy88 reviewed (approved):
beautiful !

critesjosh committed:
fix(docs): Updating info in aztec-nr docs

## Summary

Comprehensive documentation review and improvements for the aztec-nr developer documentation. This PR focuses on improving clarity, structure, accuracy, and consistency across the Aztec smart contract development guides.

### Changes Include:

**Framework & Core Concepts**
- Improved storage declaration guide with clearer examples and structure
- Enhanced macros documentation for better clarity
- Updated contract artifacts, structure, upgrades, and dependencies documentation
- Refined global variables documentation

**Function Documentation**
- Updated attributes documentation for clarity and accuracy
- Streamlined function transforms and visibility documentation

**How-To Guides**
- Enhanced event emission guide with clearer structure and examples
- Refined custom notes implementation guide
- Streamlined how-to guide for calling contracts
- Updated authwit (authentication witness) guide for clarity
- Improved cross-chain communication guide

**Advanced Topics**
- Enhanced documentation for proving historical state
- Improved retrieving and filtering notes guide
- Updated partial notes and capsules documentation
- Refactored profiling transactions guide

**L1-L2 Messaging**
- Updated L1-L2 messaging data structures documentation
- Improved inbox, outbox, and registry documentation

**Setup & Installation**
- Updated installation instructions for Noir LSP to specify VSCode Extension
- Improved local network installation instructions

**General Improvements**
- Fixed grammatical errors and typos throughout
- Improved consistency in tone and terminology
- Updated links and cross-references
- Removed redundant information and improved conciseness
- Better organization of content following simple → complex flow
